main.tf

```diff
@@ -50,7 +50,20 @@ locals {
           lookup(repo, "post_workflow_hooks", []),
           lookup(repo, "workflow", "") != "" && lookup(local._workflows, lookup(repo, "workflow", ""), "") != "" ? (
             local._workflows[lookup(repo, "workflow", "")].infracost.enabled ? [
-              { run : "infracost comment gitlab --repo $BASE_REPO_OWNER/$BASE_REPO_NAME --merge-request $PULL_NUM --path /tmp/$BASE_REPO_OWNER-$BASE_REPO_NAME-$PULL_NUM/'*'-infracost.json --gitlab-token $GITLAB_TOKEN --behavior new" }
+              { run : <<EOT
+if [ -f "/tmp/$BASE_REPO_OWNER-$BASE_REPO_NAME-$PULL_NUM/sent" ]; then
+  exit 0
+fi
+
+infracost comment gitlab --repo $BASE_REPO_OWNER/$BASE_REPO_NAME \
+                         --merge-request $PULL_NUM \
+                         --path /tmp/$BASE_REPO_OWNER-$BASE_REPO_NAME-$PULL_NUM/'*'-infracost.json \
+                         --gitlab-token $GITLAB_TOKEN \
+                         --behavior new
+
+touch /tmp/$BASE_REPO_OWNER-$BASE_REPO_NAME-$PULL_NUM/sent
+EOT
+              }
```

Comment:
The other way to achieve the same without creation of an additional flag/file would be to use the script suggested by infracost:

```yaml
      - run: |
          # post_workflow_hooks are executed after the repo workflow has run.
          # This enables you to post an Infracost comment with the combined cost output
          # from all your projects. However, post_workflow_hooks are also triggered when
          # an apply occurs. In order to stop commenting on PRs twice we need to check
          # if the Infracost output directory created in our 'plan' stage exists before continuing.
          if [ ! -d "/tmp/$BASE_REPO_OWNER-$BASE_REPO_NAME-$PULL_NUM" ]; then
            exit 0
          fi

          # Choose the commenting behavior, 'new' is a good default:
          # new: Create a new cost estimate comment on every run of Atlantis for each project.
          # update: Create a single comment and update it. The "quietest" option.
          # hide-and-new: Minimize previous comments and create a new one.
          # delete-and-new: Delete previous comments and create a new one.
          infracost comment github --repo $BASE_REPO_OWNER/$BASE_REPO_NAME \
                                   --pull-request $PULL_NUM \
                                   --path /tmp/$BASE_REPO_OWNER-$BASE_REPO_NAME-$PULL_NUM/'*'-infracost.json \
                                   --github-token $GITHUB_TOKEN \
                                   --behavior new

          # remove the Infracost output directory so that `infracost comment` is not
          # triggered on an `atlantis apply`
          rm -rf /tmp/$BASE_REPO_OWNER-$BASE_REPO_NAME-$PULL_NUM
```